File: factorial/Factorial.vb
Module Module1

    Sub Main()
        Dim n, i, f As Integer
        Console.Write("Enter a Number: ")
        n = CInt(Console.ReadLine())
        f = 1
        If n < 0 Then
            Console.WriteLine("Factorial of negative number is not possible")
        ElseIf n = 0 Or n = 1 Then
            Console.WriteLine("Factorial of " + n.ToString() + " is 1")
        Else
            For i = 1 To n
                f *= i
            Next
            Console.WriteLine("Factorial of " + n.ToString() + " is " + f.ToString())
        End If
        Console.ReadLine()
    End Sub

End Module

File: factorial/perl/factorial.pl
#!/usr/bin/perl -w

# subroutine to calculate factorial
sub getFactorial {
  my ($num) = @_;
  return 1 if $num == 0;
  return getFactorial($num-1) * $num;
}

print "Enter a number to calculate factorial: ";
chomp ($sample = <STDIN>);

if($sample == 0 or $sample == 1){
  print "Factorial of $sample is 1";
} elsif($sample < 0){
  print "Cannot calculate factorial of negative number";
} else {
  $fact = getFactorial($sample);
  print "Factorial of $sample is $fact";
}
